hack/ci/e2e.sh

```diff
@@ -37,7 +37,7 @@ install_kind() {
 main() {
   # create temp dir and setup cleanup
   TMP_DIR=$(mktemp -d)
-  trap cleanup EXIT
+  trap cleanup INT TERM EXIT
```

Comment:
this one should already be doing this because it's bash, but the other script may be ash/dash (it's using sh)

Reply:
this comment is confusing me regarding trapping EXIT
https://unix.stackexchange.com/a/149093

Reply:
> There's also what @Shawn said: Ash and Dash don't trap signals with EXIT.

that's the only relevant part really.

POSIX has requirements about what it looks like when EXIT is called, but when it is called is under specified. we explicitly want these